pandas/tests/io/xml/test_to_xml.py

```diff
@@ -202,10 +202,13 @@ def test_str_output(datapath, parser):
 
 
 def test_wrong_file_path(parser):
+    path = "/my/fake/path/output.xml"
+
     with pytest.raises(
-        FileNotFoundError, match=("No such file or directory|没有那个文件或目录")
+        OSError,
+        match=(r"Cannot save file into a non-existent directory: .*path"),
```

Comment:
Apparently in a Windows machine if you use forward-slash, `path = "/my/fake/path/output.xml"`,  `pathlib.Path(path).parent` changes it to backslash, while os.path.dirname(path) does not.

Having the backslash messes up with the regex library that pytest is using, so for the match parameter we use `*.path*` instead of the entire path `{pathlib.Path(path).parent`

Reply:
If the `os.path.dirname` solution is more reliable, it would be better to go back to it. Sorry, about that.

Reply:
I think `os.path.dirname` isn't necessarily more reliable.

The problem comes from the fact that the path being tested is hard-coded with forward-slash: https://github.com/pandas-dev/pandas/blob/ca0e751301fe7dbdfc0abe552f2d5229727b9528/pandas/tests/io/xml/test_to_xml.py#L205

So when `path` is passed into `os.path.dirname`, it stays in forward-slash even in a Windows machine. In contrast, if you pass `path` to `pathlib.Path.parent`, the forward slash becomes backslash.

Now backslash on itself is not a problem, it only becomes a problem when you pass it to the `re` module - which pytest uses. In short, the backslash has a special meaning in the `re` module (https://stackoverflow.com/a/33582461/13825658). So to have an literal `\` when using the `re` module, we need to do `\\\\`, the first escape is for the Python interpreter so you get `\\`, the second escape for the `re` module so you finally get `\`

I think it is best to avoid this altogether when using the `re` module, which is why  I use the wildcard `.*` instead passing the entire path in `pytest.raises` (line 209). It looks like it's what we do in other test cases as well, for example https://github.com/pandas-dev/pandas/blob/ca0e751301fe7dbdfc0abe552f2d5229727b9528/pandas/tests/io/test_common.py#L206

Reply:
Okay, thank you for your thorough explanation!